### ISSUE TITLE
ConcurrentModifications exceptions

### ISSUE THREAD

**Issue by martinrosstmc** [bvanalderweireldt/concurrent-unique-queue#2]
Hi,

I'm getting ConcurrentModificationExceptions while attempting to use two threads to put/take from the queue.  I do have a question around locking.  There are takeLock's and putLock's...  But I don't see how the locking preventing the LinkedHashSet from having these concurrent since they put doesn't check takeLock and viceVersa.

**Comment by bvanalderweireldt** [comment]
Yeah there is a bug, originally I wanted to avoid using a full lock for put, I'll publish a new version soon.
